File: src/main/scala/jk_5/nailed/server/map/game/script/api/ScriptScoreboardApi.scala
package jk_5.nailed.server.map.game.script.api

import jk_5.nailed.api.scoreboard.DisplayType
import jk_5.nailed.server.map.NailedMap

/**
 * No description given
 *
 * @author jk-5
 */
class ScriptScoreboardApi(private[this] val mapApi: ScriptMapApi, private[this] val map: NailedMap) {

  def getObjective(name: String) = {
    val o = map.getScoreboardManager.getOrCreateObjective(name)
    if(o == null) null else new ScriptObjectiveApi(o)
  }

  def setDisplay(objective: ScriptObjectiveApi, display: DisplayType){
    map.getScoreboardManager.setDisplay(display, map.getScoreboardManager.getObjective(objective.getObjectiveId))
  }
}

File: src/main/java/jk_5/nailed/server/map/game/script/api/ScriptScoreboardApi.java
package jk_5.nailed.server.map.game.script.api;

import jk_5.nailed.api.map.Map;
import jk_5.nailed.api.scoreboard.DisplayType;

public class ScriptScoreboardApi {

    private final ScriptMapApi scriptMapApi;
    private final Map map;

    public ScriptScoreboardApi(ScriptMapApi scriptMapApi, Map map) {
        this.scriptMapApi = scriptMapApi;
        this.map = map;
    }

    public ScriptObjectiveApi getObjective(String name){
        return new ScriptObjectiveApi(map.getScoreboardManager().getOrCreateObjective(name));
    }

    public void setDisplay(ScriptObjectiveApi objective, DisplayType display){
        map.getScoreboardManager().setDisplay(display, map.getScoreboardManager().getObjective(objective.getObjectiveId()));
    }
}
